## 경진대회~2021.01.08 17:30까지    

>1.사이킷런에 load_digits, load_breast_cancer 두가지데이터를가지고   
>2. 분류알고리즘 적용해서 정확도   
>3. 조건; test_size=0.2, random_state=2021
어떤 모델을 써서라도 최고치구하기

In [1]:
from sklearn.datasets import load_breast_cancer
cancer = load_breast_cancer()

In [2]:
dir(cancer)

['DESCR',
 'data',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

In [3]:
import pandas as pd

df = pd.DataFrame(data=cancer.data, columns=cancer.feature_names)
df.head()

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst radius  worst texture  worst perimeter  \
0                 0.07871  ...         25.38          17.33           184.60   
1                 0.05667  ...         24.99          23.41           158.80   
2                 0.05999  ...         23.57          25.53           152.50   
3                 0.09744  ...         14.91          26.50            98.87   
4                 0.05883  ...         22.54          16.67           152.20   

   worst area  worst smoothness  worst compactness  worst concavity  \
0      2019.0            0.1622             0.6656           0.7119   
1      1956.0            0.1238             0.1866           0.2416   
2      1709.0            0.1444             0.4245           0.4504   
3       567.7            0.2098             0.8663           0.6869   
4      1575.0            0.1374             0.2050           0.4000   

   worst concave points  worst symmetry  worst fractal dimension  
0                0.2654          0.4601                  0.11890  
1                0.1860          0.2750                  0.08902  
2                0.2430          0.3613                  0.08758  
3                0.2575          0.6638                  0.17300  
4                0.1625          0.2364                  0.07678  

[5 rows x 30 columns]

In [4]:
df.columns

Index(['mean radius', 'mean texture', 'mean perimeter', 'mean area',
       'mean smoothness', 'mean compactness', 'mean concavity',
       'mean concave points', 'mean symmetry', 'mean fractal dimension',
       'radius error', 'texture error', 'perimeter error', 'area error',
       'smoothness error', 'compactness error', 'concavity error',
       'concave points error', 'symmetry error', 'fractal dimension error',
       'worst radius', 'worst texture', 'worst perimeter', 'worst area',
       'worst smoothness', 'worst compactness', 'worst concavity',
       'worst concave points', 'worst symmetry', 'worst fractal dimension'],
      dtype='object')

In [5]:
#정규화
from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()
cancer_scaled = scaler.fit_transform(cancer.data)
scaled_df = pd.DataFrame(data=cancer_scaled, columns=cancer.feature_names)
scaled_df.head()

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0     0.521037      0.022658        0.545989   0.363733         0.593753   
1     0.643144      0.272574        0.615783   0.501591         0.289880   
2     0.601496      0.390260        0.595743   0.449417         0.514309   
3     0.210090      0.360839        0.233501   0.102906         0.811321   
4     0.629893      0.156578        0.630986   0.489290         0.430351   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0          0.792037        0.703140             0.731113       0.686364   
1          0.181768        0.203608             0.348757       0.379798   
2          0.431017        0.462512             0.635686       0.509596   
3          0.811361        0.565604             0.522863       0.776263   
4          0.347893        0.463918             0.518390       0.378283   

   mean fractal dimension  ...  worst radius  worst texture  worst perimeter  \
0                0.605518  ...      0.620776       0.141525         0.668310   
1                0.141323  ...      0.606901       0.303571         0.539818   
2                0.211247  ...      0.556386       0.360075         0.508442   
3                1.000000  ...      0.248310       0.385928         0.241347   
4                0.186816  ...      0.519744       0.123934         0.506948   

   worst area  worst smoothness  worst compactness  worst concavity  \
0    0.450698          0.601136           0.619292         0.568610   
1    0.435214          0.347553           0.154563         0.192971   
2    0.374508          0.483590           0.385375         0.359744   
3    0.094008          0.915472           0.814012         0.548642   
4    0.341575          0.437364           0.172415         0.319489   

   worst concave points  worst symmetry  worst fractal dimension  
0              0.912027        0.598462                 0.418864  
1              0.639175        0.233590                 0.222878  
2              0.835052        0.403706                 0.213433  
3              0.884880        1.000000                 0.773711  
4              0.558419        0.157500                 0.142595  

[5 rows x 30 columns]

In [6]:
#훈련셋나누기
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(
    cancer_scaled, cancer.target, test_size=0.2, random_state=2021
)

## 서포트벡터머신 모델 생성/학습/예측/평가

In [7]:
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score
svc = SVC()
svc.fit(X_train, y_train)
pred = svc.predict(X_test)
accuracy_score(y_test, pred)

0.9912280701754386

In [9]:
svc.get_params()

{'C': 1.0,
 'break_ties': False,
 'cache_size': 200,
 'class_weight': None,
 'coef0': 0.0,
 'decision_function_shape': 'ovr',
 'degree': 3,
 'gamma': 'scale',
 'kernel': 'rbf',
 'max_iter': -1,
 'probability': False,
 'random_state': None,
 'shrinking': True,
 'tol': 0.001,
 'verbose': False}

### SVC 모델 0.9912가 나왔다! 더 이상 만지면 나빠질 것 같다;;

In [68]:
#그래도 한번 도전~~!!
from sklearn.model_selection import GridSearchCV

params = {
    'C':[ 0.1, 1, 5, 10, 100]
}

grid_clf = GridSearchCV(svc, param_grid=params, scoring='accuracy', cv=5) 
grid_clf.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_clf.best_score_:.4f}')
print('최적 파라미터:', grid_clf.best_params_)

최고 평균 정확도: 0.9910
최적 파라미터: {'C': 5}


In [12]:
best_clf = grid_clf.best_estimator_
pred = best_clf.predict(X_test)
accuracy_score(y_test, pred)

0.9824561403508771

## 랜덤 포레스트 모델 생성/학습/예측/평가

In [8]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

rf_clf = RandomForestClassifier()
rf_clf.fit(X_train, y_train)
pred = rf_clf.predict(X_test)
acc = accuracy_score(y_test, pred)
print(f'랜덤 포레스트 정확도: {acc:.4f}')

랜덤 포레스트 정확도: 0.9649


In [13]:
#최적 파라미터 찾기
rf_clf.get_params()

{'bootstrap': True,
 'ccp_alpha': 0.0,
 'class_weight': None,
 'criterion': 'gini',
 'max_depth': None,
 'max_features': 'auto',
 'max_leaf_nodes': None,
 'max_samples': None,
 'min_impurity_decrease': 0.0,
 'min_impurity_split': None,
 'min_samples_leaf': 1,
 'min_samples_split': 2,
 'min_weight_fraction_leaf': 0.0,
 'n_estimators': 100,
 'n_jobs': None,
 'oob_score': False,
 'random_state': None,
 'verbose': 0,
 'warm_start': False}

In [14]:
from sklearn.model_selection import GridSearchCV

params = {
     'max_depth': [8, 12, 16],
     'min_samples_split': [12, 16, 20],
     'n_estimators':[ 10, 100, 200]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도0.9560로 떨어졌다. 
#파라미터 12, 12, 10

최고 평균 정확도: 0.9560
최적 파라미터: {'max_depth': 12, 'min_samples_split': 12, 'n_estimators': 10}


In [15]:
#재조정1
params = {
     'max_depth': [6, 10, 12, 14],
     'min_samples_split': [10, 11, 12, 13],
     'n_estimators':[ 6, 8, 10, 50]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도0.9582 
#파라미터 10, 13, 50

최고 평균 정확도: 0.9582
최적 파라미터: {'max_depth': 10, 'min_samples_split': 13, 'n_estimators': 50}


In [16]:
#재조정2
params = {
     'max_depth': [9, 10, 11, 15],
     'min_samples_split': [12, 13, 14],
     'n_estimators':[ 40, 50, 60, 80]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도0.9582 
#파라미터 15, 13, 80

최고 평균 정확도: 0.9582
최적 파라미터: {'max_depth': 15, 'min_samples_split': 13, 'n_estimators': 80}


In [19]:
#재조정3
params = {
     'max_depth': [10, 12, 15, 18],
     'min_samples_split': [12, 13, 14],
     'n_estimators':[ 70, 80, 90, 100]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도0.9560 
#파라미터 15, 14, 80

최고 평균 정확도: 0.9560
최적 파라미터: {'max_depth': 15, 'min_samples_split': 14, 'n_estimators': 80}


### 랜덤포레스트 결과값 0.9649인데.... 
### 그나마 최고평균 정확도 0.9582로 가장 높은 재조정1이므로   
### 하이퍼파라미터는 안만지는것이 좋다는 결론.

In [20]:
df = pd.DataFrame(grid_cv.cv_results_)
df = df[['param_n_estimators', 'param_max_depth', 'param_min_samples_split', 'mean_test_score']]
df

param_n_estimators param_max_depth param_min_samples_split  mean_test_score
0                  70              10                      12         0.945030
1                  80              10                      12         0.949431
2                  90              10                      12         0.953817
3                 100              10                      12         0.949445
4                  70              10                      13         0.949416
5                  80              10                      13         0.951638
6                  90              10                      13         0.940644
7                 100              10                      13         0.947238
8                  70              10                      14         0.951638
9                  80              10                      14         0.951638
10                 90              10                      14         0.945045
11                100              10                      14         0.951638
12                 70              12                      12         0.945045
13                 80              12                      12         0.947238
14                 90              12                      12         0.949402
15                100              12                      12         0.951638
16                 70              12                      13         0.945030
17                 80              12                      13         0.951624
18                 90              12                      13         0.945030
19                100              12                      13         0.949445
20                 70              12                      14         0.947252
21                 80              12                      14         0.949445
22                 90              12                      14         0.951624
23                100              12                      14         0.951638
24                 70              15                      12         0.949431
25                 80              15                      12         0.951638
26                 90              15                      12         0.949431
27                100              15                      12         0.945045
28                 70              15                      13         0.942837
29                 80              15                      13         0.949445
30                 90              15                      13         0.945045
31                100              15                      13         0.942837
32                 70              15                      14         0.953860
33                 80              15                      14         0.956039
34                 90              15                      14         0.947238
35                100              15                      14         0.951638
36                 70              18                      12         0.951653
37                 80              18                      12         0.949431
38                 90              18                      12         0.942837
39                100              18                      12         0.951638
40                 70              18                      13         0.951667
41                 80              18                      13         0.947238
42                 90              18                      13         0.951653
43                100              18                      13         0.947252
44                 70              18                      14         0.953831
45                 80              18                      14         0.949445
46                 90              18                      14         0.951638
47                100              18                      14         0.947223

- 튜닝된 파라미터로 재평가

In [21]:
best_clf = grid_cv.best_estimator_
pred = best_clf.predict(X_test)
acc = accuracy_score(y_test, pred)
print(f'최적 파라미터 랜덤 포레스트 정확도: {acc:.4f}')

최적 파라미터 랜덤 포레스트 정확도: 0.9561


In [22]:
# K 최근접 이웃
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier()
knn.get_params()

{'algorithm': 'auto',
 'leaf_size': 30,
 'metric': 'minkowski',
 'metric_params': None,
 'n_jobs': None,
 'n_neighbors': 5,
 'p': 2,
 'weights': 'uniform'}

In [23]:
knn.fit(X_train, y_train)
pred = knn.predict(X_test)
accuracy_score(y_test, pred)
#결과값 0.9824

0.9824561403508771

# load_digis

In [32]:
from sklearn.datasets import load_digits
digits = load_digits()

In [33]:
import pandas as pd

df = pd.DataFrame(data=digits.data, columns=digits.feature_names)
df.head()

pixel_0_0  pixel_0_1  pixel_0_2  pixel_0_3  pixel_0_4  pixel_0_5  \
0        0.0        0.0        5.0       13.0        9.0        1.0   
1        0.0        0.0        0.0       12.0       13.0        5.0   
2        0.0        0.0        0.0        4.0       15.0       12.0   
3        0.0        0.0        7.0       15.0       13.0        1.0   
4        0.0        0.0        0.0        1.0       11.0        0.0   

   pixel_0_6  pixel_0_7  pixel_1_0  pixel_1_1  ...  pixel_6_6  pixel_6_7  \
0        0.0        0.0        0.0        0.0  ...        0.0        0.0   
1        0.0        0.0        0.0        0.0  ...        0.0        0.0   
2        0.0        0.0        0.0        0.0  ...        5.0        0.0   
3        0.0        0.0        0.0        8.0  ...        9.0        0.0   
4        0.0        0.0        0.0        0.0  ...        0.0        0.0   

   pixel_7_0  pixel_7_1  pixel_7_2  pixel_7_3  pixel_7_4  pixel_7_5  \
0        0.0        0.0        6.0       13.0       10.0        0.0   
1        0.0        0.0        0.0       11.0       16.0       10.0   
2        0.0        0.0        0.0        3.0       11.0       16.0   
3        0.0        0.0        7.0       13.0       13.0        9.0   
4        0.0        0.0        0.0        2.0       16.0        4.0   

   pixel_7_6  pixel_7_7  
0        0.0        0.0  
1        0.0        0.0  
2        9.0        0.0  
3        0.0        0.0  
4        0.0        0.0  

[5 rows x 64 columns]

In [36]:
#정규화
from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()
digits_scaled = scaler.fit_transform(digits.data)
scaled_df = pd.DataFrame(data=digits_scaled, columns=digits.feature_names)
scaled_df.head()

pixel_0_0  pixel_0_1  pixel_0_2  pixel_0_3  pixel_0_4  pixel_0_5  \
0        0.0        0.0     0.3125     0.8125     0.5625     0.0625   
1        0.0        0.0     0.0000     0.7500     0.8125     0.3125   
2        0.0        0.0     0.0000     0.2500     0.9375     0.7500   
3        0.0        0.0     0.4375     0.9375     0.8125     0.0625   
4        0.0        0.0     0.0000     0.0625     0.6875     0.0000   

   pixel_0_6  pixel_0_7  pixel_1_0  pixel_1_1  ...  pixel_6_6  pixel_6_7  \
0        0.0        0.0        0.0        0.0  ...     0.0000        0.0   
1        0.0        0.0        0.0        0.0  ...     0.0000        0.0   
2        0.0        0.0        0.0        0.0  ...     0.3125        0.0   
3        0.0        0.0        0.0        0.5  ...     0.5625        0.0   
4        0.0        0.0        0.0        0.0  ...     0.0000        0.0   

   pixel_7_0  pixel_7_1  pixel_7_2  pixel_7_3  pixel_7_4  pixel_7_5  \
0        0.0        0.0     0.3750     0.8125     0.6250     0.0000   
1        0.0        0.0     0.0000     0.6875     1.0000     0.6250   
2        0.0        0.0     0.0000     0.1875     0.6875     1.0000   
3        0.0        0.0     0.4375     0.8125     0.8125     0.5625   
4        0.0        0.0     0.0000     0.1250     1.0000     0.2500   

   pixel_7_6  pixel_7_7  
0     0.0000        0.0  
1     0.0000        0.0  
2     0.5625        0.0  
3     0.0000        0.0  
4     0.0000        0.0  

[5 rows x 64 columns]

In [37]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(
    digits_scaled, digits.target, test_size=0.2, random_state=2021
)

In [40]:
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score
svc = SVC()
svc.fit(X_train, y_train)
pred = svc.predict(X_test)
accuracy_score(y_test, pred)

0.9777777777777777

In [44]:
from sklearn.model_selection import GridSearchCV

params = {
    'C':[ 0.1, 1, 4, 5, 10, 100]
}

grid_clf = GridSearchCV(svc, param_grid=params, scoring='accuracy', cv=5) 
grid_clf.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_clf.best_score_:.4f}')
print('최적 파라미터:', grid_clf.best_params_)

최고 평균 정확도: 0.9910
최적 파라미터: {'C': 4}


In [45]:
# 한번더 
from sklearn.model_selection import GridSearchCV

params = {
    'C':[ 2, 3, 4, 5]
}

grid_clf = GridSearchCV(svc, param_grid=params, scoring='accuracy', cv=5) 
grid_clf.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_clf.best_score_:.4f}')
print('최적 파라미터:', grid_clf.best_params_)
# 결과값 정확도 0.9910, 최적 파라미터: {'C': 4}

최고 평균 정확도: 0.9910
최적 파라미터: {'C': 4}


### 최적 파라미터는 'C': 4로구나~~ (정확도 0.9910)

In [46]:
best_clf = grid_clf.best_estimator_
pred = best_clf.predict(X_test)
accuracy_score(y_test, pred)

0.9805555555555555

## 랜덤 포레스트 모델 생성/학습/예측/평가

In [48]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

rf_clf = RandomForestClassifier()
rf_clf.fit(X_train, y_train)
pred = rf_clf.predict(X_test)
acc = accuracy_score(y_test, pred)
print(f'랜덤 포레스트 정확도: {acc:.4f}')

랜덤 포레스트 정확도: 0.9722


In [49]:
#최적 파라미터 찾기
rf_clf.get_params()

{'bootstrap': True,
 'ccp_alpha': 0.0,
 'class_weight': None,
 'criterion': 'gini',
 'max_depth': None,
 'max_features': 'auto',
 'max_leaf_nodes': None,
 'max_samples': None,
 'min_impurity_decrease': 0.0,
 'min_impurity_split': None,
 'min_samples_leaf': 1,
 'min_samples_split': 2,
 'min_weight_fraction_leaf': 0.0,
 'n_estimators': 100,
 'n_jobs': None,
 'oob_score': False,
 'random_state': None,
 'verbose': 0,
 'warm_start': False}

In [50]:
from sklearn.model_selection import GridSearchCV

params = {
     'max_depth': [8, 12, 16],
     'min_samples_split': [12, 16, 20],
     'n_estimators':[ 10, 100, 200]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도 0.9659
#파라미터 8, 12, 200

최고 평균 정확도: 0.9659
최적 파라미터: {'max_depth': 8, 'min_samples_split': 12, 'n_estimators': 200}


In [51]:
#재조정1
params = {
     'max_depth': [5, 7, 8, 9, 10],
     'min_samples_split': [10, 12, 13],
     'n_estimators':[ 150, 200, 300]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도 0.9687
#파라미터 10, 10, 200

최고 평균 정확도: 0.9687
최적 파라미터: {'max_depth': 10, 'min_samples_split': 10, 'n_estimators': 200}


In [52]:
#재조정2
params = {
     'max_depth': [ 8, 9, 10, 11],
     'min_samples_split': [8, 9, 10, 11, 12],
     'n_estimators':[ 180, 200, 220, 250]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도 0.9722
#파라미터 9, 8, 250

최고 평균 정확도: 0.9722
최적 파라미터: {'max_depth': 9, 'min_samples_split': 8, 'n_estimators': 250}


In [53]:
#재조정3
params = {
     'max_depth': [ 8, 9, 10],
     'min_samples_split': [7, 8, 10, 12],
     'n_estimators':[ 230, 250, 260, 270, 280]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도 0.9722
#파라미터 9, 8, 270

최고 평균 정확도: 0.9722
최적 파라미터: {'max_depth': 9, 'min_samples_split': 8, 'n_estimators': 270}


In [56]:
#재조정4- n_estimators를 찾기위하여..
params = {
     'max_depth': [ 9 ],
     'min_samples_split': [ 8 ],
     'n_estimators':[ 262, 264, 266, 268, 270, 272, 274, 276, 278]
}
rf_clf = RandomForestClassifier()
grid_cv = GridSearchCV(rf_clf, param_grid=params, cv=3)
grid_cv.fit(X_train, y_train)
print(f'최고 평균 정확도: {grid_cv.best_score_:.4f}')
print('최적 파라미터:', grid_cv.best_params_)
#결과값 정확도 0.9673
#파라미터 9, 8, 265

최고 평균 정확도: 0.9715
최적 파라미터: {'max_depth': 9, 'min_samples_split': 8, 'n_estimators': 274}


In [57]:
df = pd.DataFrame(grid_cv.cv_results_)
df = df[['param_n_estimators', 'param_max_depth', 'param_min_samples_split', 'mean_test_score']]
df

param_n_estimators param_max_depth param_min_samples_split  mean_test_score
0                262               9                       8         0.969381
1                264               9                       8         0.969381
2                266               9                       8         0.966597
3                268               9                       8         0.967989
4                270               9                       8         0.967989
5                272               9                       8         0.968685
6                274               9                       8         0.971468
7                276               9                       8         0.970772
8                278               9                       8         0.965901

- 튜닝된 파라미터로 재평가

In [58]:
best_clf = grid_cv.best_estimator_
pred = best_clf.predict(X_test)
acc = accuracy_score(y_test, pred)
print(f'최적 파라미터 랜덤 포레스트 정확도: {acc:.4f}')
#결과값0.9583

최적 파라미터 랜덤 포레스트 정확도: 0.9583


In [59]:
# K 최근접 이웃
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier()
knn.fit(X_train, y_train)
pred = knn.predict(X_test)
accuracy_score(y_test, pred)
#결과값 0.9888

0.9888888888888889